Add Scalar DB supported databases (#249)
scalar-labs/scalardb: docs/compatibility_matrix.md

> # Compatibility Matrix

Reviewer: ```suggestion
# Scalar DB Supported Databases
```

Author: Thank you for your suggestion.
Suggestion integrated.

> This document shows the compatibility of Scalar DB with supported databases.

Reviewer: ```suggestion
Scalar DB supports the following databases and the databases that are compatible with those databases.
* aaa
* bbb 
```

Author: Thank you for your suggestion.
Suggestion integrated.

> ## Scalar DB Compatibility

Reviewer: ```suggestion
## Compatibility matrix
```

Author: Thank you for your suggestion.
Suggestion integrated.

> | Scalar DB Version   | Cassandra 3.11.10 | Cassandra 3.0.24    | Cassandra 2.2.19   | Cosmos DB | DynamoDB  | MySQL 8.0.25  | MySQL 2.7.34  | PostgreSQL 13.3    | PostgreSQL 12.7   | PostgreSQL 11.12  |

Reviewer: MySQL 2.7 sounds very old.
Is it a mistake? and 5.7?

Author: Oh, sorry!
It's my mistake, 5.7 is correct.

> | Scalar DB Version   | Cassandra 3.11.10 | Cassandra 3.0.24    | Cassandra 2.2.19   | Cosmos DB | DynamoDB  | MySQL 8.0.25  | MySQL 2.7.34  | PostgreSQL 13.3    | PostgreSQL 12.7   | PostgreSQL 11.12  |

Reviewer: I'm not sure if we should note the bug-fix versions for Cassandra and MySQL. Should they be `3.11` and `8.0` respectvively?

Author: Cassandra and MySQL compatibility updated with stable versions
MySQL
```
$ brew info mysql@8.0
mysql: stable 8.0.25 (bottled)

$ brew info mysql@5.7
mysql@5.7: stable 5.7.34 (bottled) 
```
<img width="1064" alt="Screen Shot 2021-07-27 at 12 41 00 PM" src="https://user-images.githubusercontent.com/58680313/127091766-7db77a95-cf02-4ec0-a240-bd6a76305b23.png">

Author: Thank you for your review comments

|:--------------------|:------------------|:--------------------|:-------------------|:----------|:----------|:--------------|:--------------|:-------------------|:------------------|:------------------|

> | Scalar DB 3.0.0 | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ |

Reviewer 2: Do you think the table would be easier to grasp in a glance if ✓ and x were replaced by emoji such as ✅ and ❌ ?

Author: Thank you for your suggestion.
Document updated with emoji.
